cli/src/program.rs

```diff
@@ -1027,21 +1053,28 @@ fn process_program_deploy(
     } else {
         return Err("Program location required if buffer not supplied".into());
     };
-    let programdata_len = if let Some(len) = max_len {
+    let program_data_max_len = if let Some(len) = max_len {
         if program_len > len {
-            return Err("Max length specified not large enough".into());
+            return Err(
+                "Max length specified not large enough to accommodate desired program".into(),
+            );
         }
         len
     } else if is_final {
         program_len
     } else {
         program_len * 2
     };
-    let minimum_balance = rpc_client.get_minimum_balance_for_rent_exemption(
-        UpgradeableLoaderState::size_of_programdata(programdata_len),
-    )?;
 
-    let result = if do_deploy {
+    let (min_rent_exempt_program_balance, min_rent_exempt_program_data_balance) = (
```

Comment:
Sorry, why is this change for the rent-exemption of the base program included? In general, please avoid putting in unrelated changes, it makes each review much more difficult.

Reply:
> Sorry, why is this change for the rent-exemption of the base program included?

I believe this rewrite isn't changing any functionality, just a refactoring.

The original motivation was the need to accommodate `--sign-only` for `deploy`, it's not really necessary now (after we decided to ditch offline signing for `deploy`), but I was also hoping to make it a bit more readable (by grouping related stuff together) so I don't have to "untangle it" again next time I have to go through that code.

For reviewer(s) it's not as pleasant of a thing ... I get that, trying to find the right balance between those ^

Reply:
> PS: when lots of small changes touch same parts of the code it's typically more efficient and safe (no merge conflicts to resolve, can test the final result just once while aware of all the relevant context) to change it all in one fell swoop, that's why it's all here.

While it's definitely more efficient for the developer to just make all the changes at once, it's much less efficient for the maintainers because they need to deal with a refactor *and* a code change at the same time.

Those kinds of pull requests are often the most dangerous. If you look back at the bugs introduced in codebases, it's almost always because of a refactor that also changed behavior. Because they're often big, the reviewer doesn't look at it so closely, and assumes that tests will catch any issues.

While you can work however you want, when it comes to these repos, if you do a refactor, you will be asked to do it in a separate PR.